packages/@aws-cdk-testing/cli-integ/tests/cli-integ-tests/cli.integtest.ts

```diff
@@ -1064,6 +1064,46 @@ integTest(
   }),
 );
 
+integTest(
+  'cdk diff doesnt show resource metadata changes',
```

Comment:
I'm not entirely sure why, but removing resource metadata changes is the expected behavior when performing diffs with change-sets (see https://github.com/aws/aws-cdk/pull/28336), so this is a solid way to make sure the diff actually creates the change set and uses it.